### PR TITLE
Refactor templates

### PR DESCRIPTION
**Pull request by chcosta**
Trying this again now that internal builds are fixed

This is reapplying the change I reverted so that I could investigate why internal builds were failing - https://github.com/dotnet/arcade/commit/a4ef9db7c567e0a0ed066ddc0a7d5156ae55142d

I've validated via an internal build - https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=1911915&_a=summary&view=logs

### CODE REVIEW

**Comment by chcosta** [comment]
Merging, and I'll do some post-merge validation

File: eng/common/templates/phases/base.yml [hunk 1]
```diff
@@ -1,58 +1,66 @@
...
   fetchDepth: ''
-  helixSource: 'undefined_defaulted_in_base.yml'
-  helixType: 'undefined_defaulted_in_base.yml'
-  phaseName: ''
-  phase: {}
+
+  # Optional: name of the phase (not specifying phase name may cause name collisions)
+  name: ''
+
+  # Required: A defined YAML queue
```
[the remark this answers is not in the document]
Comment (chcosta) [reply eng/common/templates/phases/base.yml]: I should have included a link to the YAML queue object documentation like I did in eng/build.yml.  oops.  I'll try to update the comment at some point soon.

[the remark this answers is not in the document]
Comment (jonfortescue) [reply eng/common/templates/phases/base.yml]: no worries!

**Comment by chcosta** [comment]
Validated post-merge:

- Internal builds are passing
- CI builds are passing
- PR builds are passing
- Mirror is still functioning
- MC is still reporting telemetry

**Pull request merged**